### ISSUE TITLE
Detecting which metric to use for CPU in different environments

### ISSUE THREAD

**Issue by yaronp68** [elastic/stack-monitoring#33]
TL;DR Metricbeat ES module should report CPU correctly for both non containerised and containerised environment instead of relying on user configuration of the monitoring kibana

If possible similar solution for the current collectors / exporters

Current situation: Where the monitoring GUI assumes that the ES node runs inside a CGroup it assumes there is CPU throttling between containers and uses a combination of metrics measuring the allocated CPU versus the used CPU time. 

The detection is done using a flag in Kibana xpack monitoring settings `xpack.monitoring.ui.container.elasticsearch.enabled`
In cloud this flag is set to true by the orchestrator leading to some problems:
- ESMS clusters who are in reality on prem but monitored by a Cloud cluster are wrongly configured to be running in CGroups
- In addition, customers might get it wrong in their environments and either set this flag when they run in a non throttled CGroups environment or forget to set this flag and won't have correct calculation of CPU on the monitoring GUI

Linked issue: [https://github.com/elastic/kibana/issues/18277](https://github.com/elastic/kibana/issues/18277)
[https://github.com/elastic/x-pack-kibana/issues/3418](https://github.com/elastic/x-pack-kibana/issues/3418)
CC: @pickypg

**Comment by pickypg** [comment]
It would be ideal to have this set directly as a single, aggregateable field to avoid having the UI do any math, or any decisions in the middle of the request.

This also would dramatically simplify the experience because users don't need to understand our settings or read through documentation, and it allows for non-heterogeneous clusters (or upgraded nodes) to be reported accurate across data _buckets_ rather than just all or nothing as we have it today.

Even better, it would work on a per-cluster basis. Currently the setting controls all of the Monitoring UI. And we could walk away from that and improve the experience for things like ESMS where that _usually_ means either the local cloud monitoring CPU stats are wrong or the production cluster's CPU stats are N/A (until you go to the advanced page).

**Comment by yaronp68** [comment]
CC: @uric @zanbel  FYI
